substra/sdk/backends/local/backend.py

```diff
@@ -125,14 +125,9 @@ def __create_compute_plan_from_tuple(self, spec, key, in_tuples):
             compute_plan_id = compute_plan.compute_plan_id
             if len(in_tuples) == 0:
                 rank = 0
-            else:  # TODO Trust the user for the rank ?
-                rank = 1 + max(
-                    [
-                        in_tuple.rank
-                        for in_tuple in in_tuples
-                        if in_tuple.compute_plan_id == compute_plan_id
-                    ]
-                )
+            else:
+                # Use the rank given by the user
+                rank = spec.rank
```

Comment:
We need to have a way to compute the default rank in case none was specified.

Reply:
Good catch !